pkg/infrastructure/services/messaging_knative.go

```diff
@@ -0,0 +1,162 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package services
+
+import (
+	"fmt"
+	"github.com/RHsyseng/operator-utils/pkg/resource"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/framework"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/infrastructure"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/util"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	eventingv1 "knative.dev/eventing/pkg/apis/eventing/v1"
+	sourcesv1alpha1 "knative.dev/eventing/pkg/apis/sources/v1alpha1"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	alpha1 "knative.dev/pkg/apis/duck/v1alpha1"
+	"knative.dev/pkg/tracker"
+)
+
+const (
+	topicIdentifier = "kogito.kie.org/messageEventId"
+)
+
+// knativeMessagingDeployer implementation of messagingHandler
+type knativeMessagingDeployer struct {
+	messagingDeployer
+}
+
+func (k *knativeMessagingDeployer) createRequiredResources(service v1alpha1.KogitoService) error {
+	infra, err := k.fetchInfraDependency(service, infrastructure.IsKnativeEventingResource)
+	if err != nil || infra == nil {
+		return err
+	}
+
+	// since we depend on Knative, let's bind a SinkBinding object to our deployment
+	sinkBinding := k.newSinkBinding(service, infra)
+	if _, err := kubernetes.ResourceC(k.cli).CreateIfNotExistsForOwner(sinkBinding, service, k.scheme); err != nil {
+		return err
+	}
+
+	// fetch for consumed topics to create our triggers
+	topics, err := k.fetchRequiredTopics(service)
+	if err != nil {
+		return err
+	}
+
+	var knativeRes resource.KubernetesResource
+	exists := false
+	for _, topic := range topics {
+		if topic.Kind == consumed {
+			if exists, err = k.triggerExists(topic, service); err != nil {
+				return err
+			} else if !exists {
+				knativeRes = k.newTrigger(topic, service, infra)
+				if err := kubernetes.ResourceC(k.cli).CreateForOwner(knativeRes, service, k.scheme); err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// newTrigger creates a new Knative Eventing Trigger reference for the given Topic
+func (k *knativeMessagingDeployer) newTrigger(t messageTopic, service v1alpha1.KogitoService, infra *v1alpha1.KogitoInfra) *eventingv1.Trigger {
+	return &eventingv1.Trigger{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-listener-%s", service.GetName(), util.RandomSuffix()),
```

Comment:
rather then appending `util.RandomSuffix()` could we append `Topic` name. It will help us while fetching `Trigger` as well. Currently we are using `lable` to fetch trigger, if name are fixed then we could fetch trigger by `name` only

Reply:
Because the topic name might not be a Kubernetes resource safe name, that's why is handled by label instead.